src/main/java/teammates/common/datatransfer/NotificationTargetUser.java

```diff
@@ -0,0 +1,68 @@
+package teammates.common.datatransfer;
+
+import teammates.common.util.Const;
+
+/**
+ * Represents the type of target user groups for notifications.
+ */
+public enum NotificationTargetUser {
+
+    /**
+     * Target users are students only.
+     */
+    STUDENT(Const.NotificationTargetUser.STUDENT),
+
+    /**
+     * Target users are instructors only.
+     */
+    INSTRUCTOR(Const.NotificationTargetUser.INSTRUCTOR),
+
+    /**
+     * Target users are both instructors and students.
+     */
+    GENERAL(Const.NotificationTargetUser.GENERAL);
+
+    private final String targetUser;
+
+    NotificationTargetUser(String targetUser) {
+        this.targetUser = targetUser;
+    }
+
+    public String getTargetUser() {
+        return targetUser;
+    }
+
+    /**
+     * Formats the participant type as a singular noun.
+     *
+     * @return A user-friendly {@code String} representing this participant in singular form.
+     */
+    public String toSingularFormString() {
+        switch (this) {
+        case STUDENT:
+            return "student";
```

Comment:
Are we able to use `Const.NotificationTargetUser.STUDENT` or is there a reason to hardcode the string? Likewise for the NotificationType enum

Reply:
To take it one step further, would it be possible to simplify the `NotificationTargetUser` and `NotificationTypes` enums to something like `CommentVisibilityType`? It could potentially allow us to remove the respective enums in `Const.java` altogether as well as the functions here, thereby making the code less complicated.

Reply:
Sure, I think simplify it would be good. Previously I was making use of the enums in `Const.java` as well.